processor/routingprocessor/router.go

```diff
@@ -171,9 +173,11 @@ func (r *router[E, K]) extractExporter(name string, available map[component.ID]c
 }
 
 func (r *router[E, K]) getExporters(key string) []E {
-	e, ok := r.routes[key]
-	if !ok {
-		return r.defaultExporters
+	for _, route := range r.routes {
```

Comment:
I wonder what are the performance implications of this change. This code is in the hot path. If you are using this change in production already, are you able to provide some insights here?

Reply:
I haven't tested it in production yet, but according to the changed code, it does not break the loop behavior. Whether `routes` is a map or a slice, it always loops through all the elements in the `routes`. So, I would say it will not affect the performance negatively.

Reply:
I'm not concerned about the behavior, I'm concerned about the performance. I just double-checked, and this doesn't seem like it's in the hot path, called only on the Start path. In any case, it would make me more comfortable if we had a benchmark comparing how it was before and how it is now.